### DISCUSSION TITLE
Proper way to pass translations to the plugin

### DISCUSSION THREAD

**Discussion by adam3278** [GrapesJS/grapesjs#5860] [Q&A]
Hello,
Imagine the situation, when the plugin you wanna use is using the plugin config properties (ex. labelTitle) to make the translations work. What is the proper way to integrate it with I18 module? editor.t('key')? Thanks in advance.

**Comment by addlistener** [comment]
What I do is getting the locale from location.href & manually pass it to the config

And please refer the following https://grapesjs.com/docs/modules/I18n.html

**Comment by adam3278** [comment]
> What I do is getting the locale from location.href & manually pass it to the config
> 
> And please refer the following https://grapesjs.com/docs/modules/I18n.html

Yes, but I have the plugin that doesn’t support I18n module and is using plugin config as translation array (label prefix). What then? Docs doesn’t say anything about this situation. If you want an example for better imagination, try translating grapesjs/blocks-basic.

**Reply by addlistener** [reply]
How about copy the plugin code and change it? I guess a plugin does not contain thousand lines of code and could be managed

**Reply by artf** [reply]
Hey @adam3278 it depends on how the plugin is managing additional text and it's up to the plugin developer to give priority to the I18n module when necessary ([example](https://github.com/GrapesJS/grapesjs/blob/5aef0b9db6e46325593e53f4952345d5f9e2836c/src/block_manager/view/BlockView.ts#L143)).
As by your example, basic stuff like [blocks/categories labels](https://github.com/GrapesJS/grapesjs/blob/5aef0b9db6e46325593e53f4952345d5f9e2836c/src/i18n/locale/en.js#L12-L19) should be already easily extendable.
